### PR TITLE
feat(docs): artillery integration example

### PR DESCRIPTION
**Pull request by schoren**
This PR adds an example of how to run tracetest artillery integration

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Loom video

Add your loom video here if your work can be visualized

### CODE REVIEW

File: examples/quick-start-artillery/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Tracetest + Artillery
+
+> [Read the detailed recipe for setting up Tracetest + Artillery in our documentation.](https://docs.tracetest.io/tools-and-integrations/artillery)
+
+This is a quick start showing how you can create programs to run trace-based tests from your Artillery setup utilizing the tracetest engine and plugin. The example shows how to define the Tracetest tests, execute them, and wait for results to be ready. [Read the Typescript Quick Start Docs](https://docs.tracetest.io/tools-and-integrations/artillery) to see how to run this example.
```
Comment (adnanrahic) [inline examples/quick-start-artillery/README.md:5]:
```suggestion
This is a quick start showing how you can create programs to run trace-based tests from your Artillery setup utilizing the tracetest engine and plugin. The example shows how to define the Tracetest tests, execute them, and wait for the results to be ready.
```

File: examples/quick-start-artillery/tracetest-config.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+postgres:
```
Comment (adnanrahic) [inline examples/quick-start-artillery/tracetest-config.yaml:1]: Is this file needed with the Tracetest Agent?

File: examples/quick-start-artillery/engine-test-id.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+config:
+  target: my_target
+  tracetest:
+    token: tttoken_dfa1e954d47d1857
```
Comment (adnanrahic) [inline examples/quick-start-artillery/engine-test-id.yaml:4]: Do we keep the token here or should we replace it with `<YOUR_ENV_TOKEN>`.

File: examples/quick-start-artillery/engine-test.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+config:
+  target: my_target
+  tracetest:
+    token: tttoken_dfa1e954d47d1857
```
Comment (adnanrahic) [inline examples/quick-start-artillery/engine-test.yaml:4]: Do we keep the token here or should we replace it with `<YOUR_ENV_TOKEN>`.

File: examples/quick-start-artillery/plugin-test-id.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            type: "Load test"
+
+        traces:
+          reporter: otlp-http
+          endpoint: "http://localhost:4318/v1/traces"
+          attributes:
+            environment: "test"
+            tool: "Artillery"
+    tracetest:
+      token: tttoken_dfa1e954d47d1857
```
Comment (adnanrahic) [inline examples/quick-start-artillery/plugin-test-id.yaml:25]: Do we keep the token here or should we replace it with `<YOUR_ENV_TOKEN>`.

File: examples/quick-start-artillery/plugin-test.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            type: "Load test"
+
+        traces:
+          reporter: otlp-http
+          endpoint: "http://localhost:4318/v1/traces"
+          attributes:
+            environment: "test"
+            tool: "Artillery"
+    tracetest:
+      token: tttoken_dfa1e954d47d1857
```
Comment (adnanrahic) [inline examples/quick-start-artillery/plugin-test.yaml:25]: Do we keep the token here or should we replace it with `<YOUR_ENV_TOKEN>`.

File: examples/quick-start-artillery/tracetest-provision.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (adnanrahic) [inline examples/quick-start-artillery/tracetest-provision.yaml:1]: Does the user need to run a command with the CLI to configure the provision or will it happen on-startup?

eg.

```
tracetest apply -f ./tracetest-provision.yaml
```

If it is needed, we might need to specify this?

File: examples/quick-start-artillery/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,147 @@
+version: "3.5"
+name: pokeshop
+
+services:
+  tracetest-agent:
+    environment:
+      TRACETEST_DEV: ${TRACETEST_DEV}
+      TRACETEST_API_KEY: ${TRACETEST_AGENT_API_KEY}
+      TRACETEST_SERVER_URL: ${TRACETEST_SERVER_URL}
```
Comment (adnanrahic) [inline examples/quick-start-artillery/docker-compose.yml:9]: This is not specified in the .env file. Should we keep or remove it? It might confuse users.

File: examples/quick-start-artillery/.env.template [hunk 9]
```diff
@@ -0,0 +1 @@
+TRACETEST_AGENT_API_KEY=
```
Comment (adnanrahic) [inline examples/quick-start-artillery/.env.template:1]: Would it be possible to add the `TRACETEST_API_TOKEN` here as well and load it in the yaml files? This could mean removing the hardcoded token values from the yamls and making sure the user only sets it once.

**Review by adnanrahic — COMMENTED** [review]
Awesome work! Added some suggestions and comments.

**Comment by adnanrahic** [comment]
Also, @xoscar -- Are you planning on writing the docs recipe in this PR to be released together? I can help with that if needed.

**Review by jfermi — APPROVED** [review]

**Review by xoscar — APPROVED** [review]
